blackboxopt/base.py

```diff
@@ -133,14 +148,23 @@ def __init__(
         super().__init__(search_space=search_space, seed=seed)
         self.objective = objective
 
-    def report_evaluation(self, evaluation: Evaluation) -> None:
-        raise_on_unknown_or_incomplete(
-            exception=ObjectivesError,
-            known=[self.objective.name],
-            reported=evaluation.objectives.keys(),
-        )
+    def report(self, evaluations: Union[Evaluation, Iterable[Evaluation]]) -> None:
```

Comment:
Let's introduce a docstring to explain the behaviour when erroneous evaluations are reported, which ones are accepted and how the user gets to learn about the failed ones and what went wrong for them (incl. a "raises" docstring section).